src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -387,40 +464,56 @@ const useGroupInfo = (channelId = null) => {
       await handleMessageAnonymously();
     }
   };
-  const onLeaveGroup = () => {
-    Alert.alert('', 'Exit this group?', [{text: 'Cancel'}, {text: 'Exit', onPress: leaveGroup}]);
-  };
 
-  const leaveGroup = async () => {
+  const actionLeaveGroup = async () => {
+    setOpenModal(false);
+    const responseChannelData = await leaveGroup({channelId});
+    console.warn('responseChannelData', responseChannelData);
     try {
-      const response = await channel.removeMembers([profile.myProfile.user_id]);
-      SimpleToast.show('You left this chat');
-      navigation.reset({
-        index: 1,
-        routes: [
-          {
-            name: 'AuthenticatedStack',
-            params: {
-              screen: 'HomeTabs',
-              params: {
-                screen: 'ChannelList'
-              }
-            }
-          }
-        ]
-      });
-      setNewParticipant(response.members);
+      const {channelName} = getChannelListInfo(
+        responseChannelData.data,
+        signedProfileId,
+        anonProfileId
+      );
+      const channelList = await ChannelListSchema.getSchemaById(localDb, channelId);
+      channelList.name = channelName;
+      channelList.rawJson = responseChannelData.data;
+      await channelList.save(localDb);
+
+      setSelectedChannel(channelList);
+
+      await UserSchema.deleteByUserId(localDb, signedProfileId, channelId);
     } catch (e) {
-      console.log(e, 'sayu');
+      if (__DEV__) {
+        console.log(e, 'error');
+      }
+      console.log('error on memberSchema');
+      console.log(JSON.stringify(e));
+      console.log(e);
     }
+
+    refresh('channelList');
+    refresh('chat');
+    refresh('channelInfo');
+
+    setTimeout(() => {
+      navigation.navigate('HomeTabs');
+    }, 500);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `actionLeaveGroup` function has been added to handle leaving the group chat. It uses the `leaveGroup` service and updates the local database accordingly. Again, consider replacing the console logs with a proper logging service.
```diff
- console.log('error on memberSchema');
- console.log(JSON.stringify(e));
- console.log(e);
+ // Consider replacing the above lines with a logging service
+ logger.error('Error on memberSchema', e);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->